packages/medusa/src/api/routes/store/products/list-products.ts

```diff
@@ -77,9 +78,9 @@ export default async (req, res) => {
   // get only published products for store endpoint
   filterableFields["status"] = ["published"]
 
-  let includeFields: string[] = []
+  let includeFields: (keyof Product)[] = []
   if (validated.fields) {
-    const set = new Set(validated.fields.split(","))
+    const set = new Set(validated.fields.split(",")) as Set<keyof Product>
```

Comment:
You have a point! I think the issue is coming from the fact that the `listAndCount` method takes in a `FindProductConfig` type which uses the `FindConfig<Entity>` utility type:

```typescript
export interface FindConfig<Entity> {
  select?: (keyof Entity)[] // whoops
  skip?: number
  take?: number
  relations?: string[]
  order?: Record<string, "ASC" | "DESC">
}
```

Should we change this one? I think this one might affect the typings in different places so i'm a bit reluctant 😅. As an alternative I could change the `FindProductConfig` type like this:
```typescript
type FindProductConfig = Omit<FindConfig<Product>, "select"> & {
  select?: string[]
} & PriceListLoadConfig
```

Wdyt?

Reply:
in that case we can say (for the moment) that select is `(keyof Entity)[] | string[]` wdyt?

Reply:
Unfortunately that also breaks a few things esp. with regards to typeorm's repo methods like `findOne` 😅 . Consider this:
```typescript
class PriceListService {
  async retrieve(
    priceListId: string,
    config: FindConfig<PriceList> = {}
  ): Promise<PriceList> {
    const priceListRepo = this.manager_.getCustomRepository(this.priceListRepo_)

    const query = buildQuery({ id: priceListId }, config)
    const priceList = await priceListRepo.findOne(query) // error: string[] is not assignable to (keyof PriceList)[] .. this is because findOne takes in an argument of type FindOneOptions (see below)
    // ...
  }
}
```
```typescript
// from typeorm
export interface FindOneOptions<Entity = any> {
    /**
     * Specifies what columns should be retrieved.
     */
    select?: (keyof Entity)[];
    
    // ...
}
```

Reply:
Oouch, but what it looks like when you select relations column then?

Reply:
ok, so keep the casting, but you see what I mean here :)